src/main/java/jcifs/Config.java

```diff
@@ -0,0 +1,203 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * 
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+package jcifs;
+
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Properties;
+import java.util.StringTokenizer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import jcifs.context.SingletonContext;
+
+
+/**
+ * This class now contains only utilities for config parsing.
+ * 
+ * We strongly suggest that you create an explicit {@link jcifs.context.CIFSContextWrapper}
+ * with your desired config. It's base implementation {@link jcifs.context.BaseContext}
+ * should be sufficient for most needs.
+ * 
+ * If you want to retain the classic singleton behavior you can use
+ * {@link jcifs.context.SingletonContext#getInstance()}
+ * witch is initialized using system properties.
+ * 
+ */
+@SuppressWarnings ( "javadoc" )
+public class Config {
+
+    private static final Logger log = LoggerFactory.getLogger(Config.class);
+
+
+    /**
+     * This static method registers the SMB URL protocol handler which is
+     * required to use SMB URLs with the <tt>java.net.URL</tt> class. If this
+     * method is not called before attempting to create an SMB URL with the
+     * URL class the following exception will occur:
+     * <blockquote>
+     * 
+     * <pre>
+     * Exception MalformedURLException: unknown protocol: smb
+     *     at java.net.URL.&lt;init&gt;(URL.java:480)
+     *     at java.net.URL.&lt;init&gt;(URL.java:376)
+     *     at java.net.URL.&lt;init&gt;(URL.java:330)
+     *     at jcifs.smb.SmbFile.&lt;init&gt;(SmbFile.java:355)
+     *     ...
+     * </pre>
+     * 
+     * <blockquote>
+     */
+    public static void registerSmbURLHandler () {
+        SingletonContext.registerSmbURLHandler();
+    }
+
+
+    /**
+     * Retrieve an <code>int</code>. If the key does not exist or
+     * cannot be converted to an <code>int</code>, the provided default
+     * argument will be returned.
+     */
+    public static int getInt ( Properties props, String key, int def ) {
+        String s = props.getProperty(key);
+        if ( s != null ) {
+            try {
+                def = Integer.parseInt(s);
+            }
+            catch ( NumberFormatException nfe ) {
+                log.error("Not a number", nfe);
+            }
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 81:88, 99:106.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.